docs: mention possibility of prerendering api routes (#31234)
nuxt/nuxt: docs/3.api/3.utils/prerender-routes.md

::note
In the browser, or if called outside prerendering, `prerenderRoutes` will have no effect.
::

> You can even prerender API routes which is particularly useful for full Statically Generated Sites (SSG) because you can then `$fetch` data as if you have an available server!

Reviewer: ```suggestion
You can even prerender API routes which is particularly useful for full statically generated sites (SSG) because you can then `$fetch` data as if you have an available server!
```

```js

> prerenderRoutes('/api/content/article/name-of-article');

Reviewer: ```suggestion
prerenderRoutes('/api/content/article/name-of-article')
```

// Somewhere later in App
const articleContent = await $fetch('/api/content/article/name-of-article', {
  responseType: 'json',

> });

Reviewer: ```suggestion
})
```

```

::warning

> Prerendered API routes in production may not return the expected response headers, depending on the provider you deploy to. For example, a JSON response might be served with an `application/octet-stream` content type. In this case, you may have to specify an explicit `responseType`.

Reviewer: ```suggestion
Prerendered API routes in production may not return the expected response headers, depending on the provider you deploy to. For example, a JSON response might be served with an `application/octet-stream` content type.
```

> Always manually set `responseType` when fetching prerenderered API routes.

Reviewer: ```suggestion
Always manually set `responseType` when fetching prerendered API routes.
```

> Fetching prerendered API routes on production looses all headers you set up in `defineEventHandler` on server side! This means the data you fetch will most likely have `application/octet-stream` content type even if it is just a JSON and therefore break the application in unpredictable ways.

Reviewer: ```suggestion
Prerendered API routes in production may not return the expected response headers, depending on the provider you deploy to. For example, a JSON response might be served with an `application/octet-stream` content type. In this case, you may have to specify an explicit `responseType`.
```

> Always manually set `responseType` when fetching prerenderered API routes!

Reviewer: ```suggestion
Always manually set `responseType` when fetching prerenderered API routes.
```
